src/choose-location/choose-location.component.test.tsx

```diff
@@ -1,158 +1,57 @@
-    });
-    act(() => {
-      fireEvent.click(marsInput);
-    });
-    await wait(() => {
-      expect(submitButton).not.toHaveAttribute("disabled");
-    });
-  });
-
-  it(`makes an API request when you submit the form`, async () => {
-    expect(setSessionLocation).not.toHaveBeenCalled();
-    act(() => {
-      fireEvent.change(searchInput, { target: { value: "Mars" } });
-    });
-    await wait(() => {
-      expect(wrapper.queryByText("Mars")).not.toBeNull();
-      marsInput = wrapper.getByLabelText("Mars");
-    });
-    fireEvent.click(marsInput);
-    fireEvent.click(submitButton);
-    await wait(() => expect(setSessionLocation).toHaveBeenCalled());
-  });
-
-  it(`send the user to the home page on submit`, async () => {
-    expect(setSessionLocation).not.toHaveBeenCalled();
-    act(() => {
-      fireEvent.change(searchInput, { target: { value: "Mars" } });
-    });
-    await wait(() => {
-      expect(wrapper.queryByText("Mars")).not.toBeNull();
-      marsInput = wrapper.getByLabelText("Mars");
-    });
-    fireEvent.click(marsInput);
-    fireEvent.click(submitButton);
-    await wait(() => {
-      expect(historyMock.push.mock.calls.length).toBe(1);
-      expect(historyMock.push.mock.calls[0][0]).toBe("/home");
-    });
-  });
-
-  it(`send the user to the redirect page on submit`, async () => {
-    cleanup();
-    mockSearch.mockResolvedValue(loginLocations);
 
+  it(`should redirect back to referring page on successful login when there is only one location`, async () => {
     const locationMock = {
       state: {
         referrer: "/home/patient-search",
       },
     };
-    wrapper = render(
-      <ChooseLocation
-        history={historyMock}
-        loginLocations={locationEntries}
-        location={locationMock}
-      />
-    );
-    searchInput = wrapper.container.querySelector("input");
-    act(() => {
-      fireEvent.change(searchInput, { target: { value: "Mars" } });
-    });
-    await wait(() => {
-      expect(wrapper.queryByText("Mars")).not.toBeNull();
-      marsInput = wrapper.getByLabelText("Mars");
-    });
-    submitButton = wrapper.getByText("Confirm", { selector: "button" });
-    expect(setSessionLocation).not.toHaveBeenCalled();
-    fireEvent.click(marsInput);
-    fireEvent.click(submitButton);
-    await wait(() => {
-      expect(historyMock.push.mock.calls.length).toBe(1);
-      expect(historyMock.push.mock.calls[0][0]).toBe(
-        locationMock.state.referrer
-      );
+    cleanup();
+    const wrapper = renderWithRouter(ChooseLocation, {
+      location: locationMock,
     });
+    await act(wait);
+    expect(wrapper.history.location.pathname).toBe(locationMock.state.referrer);
+  });
+
+  it(`should set location and skip location select page if there is exactly one location`, async () => {
+    cleanup();
+    const wrapper = renderWithRouter(ChooseLocation, {});
+    await act(wait);
+    expect(wrapper.history.location.pathname).toBe("/home");
   });
 });
```

Comment:
I'd like to also test that
- If there's more than one location and chooseLocation is enabled, the location chooser gets rendered
- If chooseLocation is disabled then the user skips the choose location page

Reply:
Added!